src/main/java/org/kuali/kfs/module/cam/web/struts/CapitalAssetInformationAction.java

```diff
@@ -103,6 +110,64 @@ protected void prepareRecordsForDisplay(CapitalAssetInformationForm capitalAsset
         capitalAssetForm.setCapitalAssetInformation(capitalAssetInformation);
     }
 
+    private boolean shouldBuildMissingCapitalAssetInformation(List<CapitalAssetInformation> capitalAssetInformations,
+            Collection<GeneralLedgerEntry> glEntries) {
+        boolean isCapitalAssetInformationAndGLEntriesDifferentSizes = !capitalAssetInformations.isEmpty() && capitalAssetInformations.size() != glEntries.size();
+        if (isGeneralErrorCorrectionDocument(glEntries)) {
+            return isCapitalAssetInformationAndGLEntriesDifferentSizes && isCapitalAssetInformationAndGLEntryTotalsDifferent(capitalAssetInformations, glEntries);
+        } else {
+            return isCapitalAssetInformationAndGLEntriesDifferentSizes;
+        }
+    }
+    
+    private boolean isGeneralErrorCorrectionDocument(Collection<GeneralLedgerEntry> glEntries) {
+        boolean isGeneralErrorCorrection = false;
+        if (CollectionUtils.isNotEmpty(glEntries)) {
+            GeneralLedgerEntry firstEntry = glEntries.iterator().next();
+            isGeneralErrorCorrection = StringUtils.equalsIgnoreCase(firstEntry.getFinancialDocumentTypeCode(), 
+                    KFSConstants.FinancialDocumentTypeCodes.GENERAL_ERROR_CORRECTION);
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("isGeneralErrorCorrectionDocument: " + isGeneralErrorCorrection);
+        }
+        return isGeneralErrorCorrection;
+    }
+    
+    private boolean isCapitalAssetInformationAndGLEntryTotalsDifferent(List<CapitalAssetInformation> capitalAssetInformations,
+            Collection<GeneralLedgerEntry> glEntries) {
+        KualiDecimal entriesTotal = findTotalGLEntries(glEntries);
+        KualiDecimal informationTotal = findTotalAmountForAssetInformation(capitalAssetInformations);
+        if (LOG.isDebugEnabled()) {
```

Comment:
I think the code would be cleaner and more clear if you remove the debug logging.

Reply:
How about moving these logs to a separate function, that will keep this function cleaner, I think this log statement will be helpful